### PR TITLE
OBPIH-3838 Fix issues with gl account for order adjustments in invoice

### PR DESCRIPTION
**Pull request by DominikaHatala**

### CODE REVIEW

File: grails-app/services/org/pih/warehouse/invoice/InvoiceService.groovy [hunk 1]
```diff
@@ -337,7 +337,7 @@ class InvoiceService {
         InvoiceItem invoiceItem = new InvoiceItem(
             budgetCode: orderAdjustment.budgetCode,
             product: orderAdjustment.orderItem?.product,
-            glAccount: orderAdjustment.glAccount ?: orderAdjustment.orderItem?.glAccount,
+            glAccount: orderAdjustment.orderItem?.glAccount ?: orderAdjustment.orderAdjustmentType?.glAccount,
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/invoice/InvoiceService.groovy:340]: This doesn't feel right. It seems the order of precedence should be 

order adjustment (override) -> order item (override) -> order adjustment type (default)

Comment (DominikaHatala) [reply grails-app/services/org/pih/warehouse/invoice/InvoiceService.groovy]: @jmiranda You're right, I've missed that. I updated PR.

Comment (jmiranda) [reply grails-app/services/org/pih/warehouse/invoice/InvoiceService.groovy]: @awalkowiak @DominikaHatala If the ticket says to do it the way you did it, then ignore my comments and merge it. If it's not clear then go with the precedence order that I specified.

```
orderAdjustment?.glAccount ?: orderAdjustment?.orderItem?.glAccount ?: orderAdjustment?.orderAdjustmentType?.glAccount

```

Comment (DominikaHatala) [reply grails-app/services/org/pih/warehouse/invoice/InvoiceService.groovy]: It was not clearly specified in the ticket so I applied your suggestions.

**Review by jmiranda — APPROVED** [review]

**Pull request merged**